### PR TITLE
Chapter 3: Mostly static pages & Chapter 4: Rails-flavored Ruby

### PR DESCRIPTION
**Pull request by baaohuii**

### CODE REVIEW

**Comment by baaohuii** [comment]
ready

File: app/assets/stylesheets/static_page.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the StaticPage controller here.
```
Comment (saobv-2223) [inline app/assets/stylesheets/static_page.scss]: remove file that never used

File: app/channels/application_cable/channel.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 module ApplicationCable
+
```
Comment (saobv-2223) [inline app/channels/application_cable/channel.rb]: remove blank line

File: app/channels/application_cable/connection.rb [hunk 3]
```diff
@@ -1,4 +1,6 @@
 module ApplicationCable
+
```
Comment (saobv-2223) [inline app/channels/application_cable/connection.rb]: remove blank line

File: app/controllers/chapter_3_controller.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class Chapter3Controller < ApplicationController
+	
+  def ... ; end
+
+end
```
Comment (saobv-2223) [inline app/controllers/chapter_3_controller.rb]: remove this file, please

File: app/controllers/static_page_controller.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class StaticPageController < ApplicationController
+  def home
+  end
+
+  def help
+  end
```
Comment (saobv-2223) [inline app/controllers/static_page_controller.rb]:
```suggestion
  def home; end

  def help; end
```

File: app/helpers/application_helper.rb [hunk 6]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+    def full_title page_title = ""
+        base_title = "Ruby on Rails Tutorial Sample App"
+        page_title.empty? ? base_title : "#{page_title} | #{base_title}"
+    end
```
Comment (saobv-2223) [inline app/helpers/application_helper.rb]: indenting only 2 spaces, please

File: app/helpers/static_page_helper.rb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+module StaticPageHelper
+end
```
Comment (saobv-2223) [inline app/helpers/static_page_helper.rb]: remove file that never used

File: app/views/layouts/application.html.erb [hunk 8]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <title>SampleApp</title>
+   <title><%= full_title yield :title %></title>
```
Comment (saobv-2223) [inline app/views/layouts/application.html.erb]:
```suggestion
   <title><%= full_title yield(:title) %></title>
```

File: app/views/static_page/help.html.erb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+<%= provide:title, "Help" %>
+
+<h1>Help</h1>
+<p>
+ Get help on the Ruby on Rails Tutorial at the
+ <a href="https://www.railstutorial.org/help">Rails Tutorial Help page</a>.
+ To get help on this sample app, see the
+ <a href="https://www.railstutorial.org/book"><em>Ruby on Rails Tutorial</em> book</a>.
```
Comment (saobv-2223) [inline app/views/static_page/help.html.erb]: using link_to, please

File: app/views/static_page/home.html.erb [hunk 10]
```diff
@@ -0,0 +1,8 @@
+<%= provide :title, "Home" %>
+
+<h1>Sample App</h1>
+<p>
+ This is the home page for the
+ <a href="https://www.railstutorial.org/">Ruby on Rails Tutorial</a>
```
Comment (saobv-2223) [inline app/views/static_page/home.html.erb]: using link_to instead of a tag

**Review by saobv-2223 — COMMENTED** [review]
Don't push cache folder to git
1. Open `.gitignore` file and add `cache`
2. Run command `git rm --cached -r cache`

Your pull request is have 2 commits, merge them to 1 commit

**Comment by baaohuii** [comment]
ready

File: app/helpers/application_helper.rb [hunk 11]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+ def full_title page_title = ""
+  base_title = "Ruby on Rails Tutorial Sample App"
+  page_title.empty? ? base_title : "#{page_title} | #{base_title}"
+ end
```
Comment (saobv-2223) [inline app/helpers/application_helper.rb]:
```suggestion
 def full_title page_title
  base_title = "Ruby on Rails Tutorial Sample App"
  page_title.blank? ? base_title : "#{page_title} | #{base_title}"
 end
```

File: app/views/static_page/help.html.erb [hunk 12]
```diff
@@ -0,0 +1,9 @@
+<%= provide:title, "Help" %>
+
+<h1>Help</h1>
+<p>
+ Get help on the Ruby on Rails Tutorial at the
+ <%= link_to "Rails Tutorial Help page", "https://www.railstutorial.org/help" %>
```
Comment (saobv-2223) [inline app/views/static_page/help.html.erb]: with the link, you should use gem config to manage them. 
https://github.com/rubyconfig/config

File: app/views/static_page/home.html.erb [hunk 13]
```diff
@@ -0,0 +1,8 @@
+<%= provide :title, "Home" %>
+
+<h1>Sample App</h1>
+<p>
+ This is the home page for the
+ <%= link_to "Ruby on Rails Tutorial", "https://www.railstutorial.org/" %>
+ sample application.
```
Comment (saobv-2223) [inline app/views/static_page/home.html.erb]:
```suggestion
  This is the home page for the
  <%= link_to "Ruby on Rails Tutorial", "https://www.railstutorial.org/" %>
  sample application.
``` and do same as above issue

File: config/database.yml [hunk 14]
```diff
@@ -0,0 +1,22 @@
+default: &default
```
Comment (saobv-2223) [inline config/database.yml]: 1. Open `.gitignore` and add `config/database.yml`
2. Run command: `git rm --cached config/database.yml`

**Comment by baaohuii** [comment]
ready

File: app/assets/config/manifest.js [hunk 15]
```diff
@@ -1,2 +1,2 @@
-//= link_tree ../images
 //= link_directory ../stylesheets .css
+//= link application.css
```
Comment (saobv-2223) [inline app/assets/config/manifest.js]: image

**Review by saobv-2223 — COMMENTED** [review]
fix comment and merge

**Pull request merged**